### PR TITLE
Add cddl installation instruction

### PR DESCRIPTION
**Pull request by mingpeiwk**

### CODE REVIEW

**Comment by mingpeiwk** [comment]
Note that this PR doesn't affect draft publish.

**Comment by mingpeiwk** [comment]
The Anweiss CDDL tool has different usage from that of Ruby cddl. The former needs cddl input after commands while the later requires cddl file input immediately after the command cddl.

**Gem version:**

cddl spec.cddl validate file.cbor

**Anweiss Rust version:**

cddl validate --cddl spec.cddl --cbor file.cbor

The Makefile under cddl folder uses the Ruby gem download version.

File: README.md [hunk 1]
```diff
@@ -28,6 +28,13 @@ apt-get -y install ruby-kramdown-rfc2629 python3-pip git
 pip3 install xml2rfc
 ```
 
+Installing cddl tool https://rubygems.org/gems/cddl
+```
+gem install cddl
```
Comment (dthaler) [inline README.md:33]: After running this command, I get

"/var/lib/gems/2.7.0/gems/cddl-0.9.1/lib/cddl.rb:1262:in `type_recall': Unknown type suit-reference nil [{}, {}, {}, {}, {}] (RuntimeError)"

Comment (mingpeiwk) [reply README.md:33]: I was able to run this in my Mac laptop (MACOS 12.6), and also earlier version by -v option. My Mac has Ruby Gems v2.6.0.

BTW, the validation of CBOR files from the Makefile generates an error in using SUIT_*, which isn't in scope for this PR to fix. We need a different PR to fix the schema reference to SUIT.

cddl check-draft-ietf-teep-protocol.cddl validate ../cbor/query_request.diag.bin || exit 1; cddl check-draft-ietf-teep-protocol.cddl validate ../cbor/query_response.diag.bin || exit 1; cddl check-draft-ietf-teep-protocol.cddl validate ../cbor/update.diag.bin || exit 1; cddl check-draft-ietf-teep-protocol.cddl validate ../cbor/teep_success.diag.bin || exit 1; cddl check-draft-ietf-teep-protocol.cddl validate ../cbor/teep_error.diag.bin || exit 1;
/Library/Ruby/Gems/2.6.0/gems/cddl-0.9.1/lib/cddl.rb:1077:in **`rule_lookup': $$SUIT_Parameters not a type** [:grpchoice, [[:member, 1, 1, [:int, 1], [:type1, [:anno, :size, [:prim, 2], [:int, 16]], [:prim, 6, 112, [:prim, 2]]]]], [[:member, 1, 1, [:int, 2], [:anno, :size, [:prim, 2], [:int, 16]]]], [[:member, 1, 1, [:int, 3], [:anno, :cbor, [:prim, 2], [:array, [:member, 1, 1, [:text, "suit-digest-algorithm-id"], [:type1, [:int, -16], [:int, -18], [:int, -43], [:int, -44], [:int, -45]]], [:member, 1, 1, [:text, "suit-digest-bytes"], [:prim, 2]]]]]], [[:member, 1, 1, [:int, 14], [:prim, 0]]], [[:member, 1, 1, [:int, 5], [:prim, 0]]], [[:member, 1, 1, [:int, 21], [:prim, 3]]], [[:member, 1, 1, [:int, 22], [:prim, 0]]], [[:member, 1, 1, [:int, 23], [:prim, 2]]], [[:member, 1, 1, [:int, 24], [:anno, :size, [:prim, 2], [:int, 16]]]], [[:member, 1, 1, [:prim, 1], [:type1, [:type1, [:prim, 0], [:prim, 1]], [:type1, [:prim, 7, 20], [:prim, 7, 21]], [:prim, 3], [:prim, 2]]]], [[:member, 1, 1, [:int, 18], [:prim, 2]]], [[:member, 1, 1, [:int, 12], [:type1, [:prim, 7, 20], [:prim, 7, 21]]]], [[:member, 1, 1, [:int, 13], [:type1, [:prim, 7, 20], [:prim, 7, 21]]]]] (**RuntimeError**)
	from /Library/Ruby/Gems/2.6.0/gems/cddl-0.9.1/lib/cddl.rb:1259:in `type_recall'

Comment (mingpeiwk) [reply README.md:33]: I got the same error Dave you got in cddl validation after I updated the suit-report cddl to its latest spec.

cddl check-draft-ietf-teep-protocol.cddl validate ../cbor/query_request.diag.bin || exit 1; cddl check-draft-ietf-teep-protocol.cddl validate ../cbor/query_response.diag.bin || exit 1; cddl check-draft-ietf-teep-protocol.cddl validate ../cbor/update.diag.bin || exit 1; cddl check-draft-ietf-teep-protocol.cddl validate ../cbor/teep_success.diag.bin || exit 1; cddl check-draft-ietf-teep-protocol.cddl validate ../cbor/teep_error.diag.bin || exit 1;
/Library/Ruby/Gems/2.6.0/gems/cddl-0.9.1/lib/cddl.rb:1262:in `type_recall': Unknown type suit-reference nil [{}, {}, {}, {}, {}] (RuntimeError)

**Comment by mingpeiwk** [comment]
After removing a line break in draft-ietf-suite-manifest.cddl for the following definition:

$$SUIT_Parameters //= (suit-parameter-vendor-identifier => 
  (RFC4122_UUID / cbor-pen))

that anweiss cddl compiler reports as non-conformant, the cddl files become conformant. The Make in cddl folder can proceed. There is message to schema match errors, but this has moved forward.

CDDL validation failure (nil for [1, {20=>"\xA0\xA1\xA2\xA3\xA4\xA5\xA6\xA7\xA8\xA9\xAA\xAB\xAC\xAD\xAE\xAF", 3=>[0]}, [[[18, -7]], [[18, -8]]], 3]):

**Review by dthaler — APPROVED** [review]

**Pull request merged**